Makefile: add ability to enable qla-32gbit target

Hi-Angel opened the pull request:
Although qla-32gbit is mentioned in the README, but there wasn't really a
way to enable it sans editing the code.

This commit adds an ability to enable the driver by passing
`QLA_32GBIT=y` variable at build time.

-----------

Worth noting, apparently Makefile build system doesn't support combining `ifdef` with `ifeq`s by using some `and`-like syntax, so I had to duplicate the `QLA_INI_DIR=qla2x00t` assignment into two different branches. Better ideas are welcome.

bvanassche commented:
There may be more users today who use the qla2x00t-32gbit driver than the original qla2x00t driver. How about changing the Makefile such that the qla2x00t-32gbit driver is built by default and such that the qla2x00t driver is only built upon explicit request?

Hi-Angel commented:
Sure, no problem! Will do

Hi-Angel commented:
Hmm, shall I add it as a separate commit? That is, so it would be easier to revert if something goes wrong. Or shall I put it into the same commit that added the ability to switch between the drivers?

bvanassche commented:
One commit per change please. That makes reviewing easier.

Hi-Angel commented:
Thanks! Done as a new commit. 

Btw, I like that, it simplified the code and the duplicate condition in Makefile now is gone.

Please see if you are okay with this change.

Hi-Angel commented:
**UPD**: rebased on latest master, fixed conflicts

Hi-Angel commented:
> **UPD**: rebased on latest master, fixed conflicts

Btw, I think [in this commit](https://github.com/bvanassche/scst/commit/237910043bffc4ac23c61cf6d803d5d3be22025c) instead of `uname -r` should've been used `$(KVER)` variable

bvanassche commented:
$(uname -r) has been changed into $(KVER) in debian/rules. Thanks for the patches!

Hi-Angel commented:
Thank you too!
